### PR TITLE
Rename isupper, islower, ucfirst and lcfirst

### PR DESCRIPTION
**Pull request by nalimilan**
Spell "uppercase" and "lowercase" in full for consistency with `uppercase` and `lowercase` functions.

The current function names seem to be inspired from different languages (Python and Perl), and are therefore inconsistent. Here's a small survey of what other languages use:

This PR:| uppercase | lowercase | isuppercase | islowercase | uppercasefirst | lowercasefirst
-----------|----------------|----------------|-----------------|-----------------|---------------------|-------------------
Python  | upper         | lower          | isupper        | islower         | capitalize           |
Ruby     | upcase       | downcase |                    |                       | capitalize           |
Perl       | uc               | lc               |                    |                       |  ucfirst               | lcfirst
Java     | toUpperCase | toLowerCase | isUpperCase | isLowerCase         |  StringUtils.capitalize  |  StringUtils.uncapitalize
Rust     | to_uppercase | to_lowercase | is_uppercase | is_lowercase         |                         |
Swift     | uppercased | lowercased | long* | long*         |                       |
R          | toupper         | tolower        |        |                   | Hmisc::capitalize               |

 \* long: `CharacterSet.uppercaseLetters.contains(c)` and `CharacterSet.lowercaseLetters.contains(c)`.

### CODE REVIEW

**Comment by JeffBezanson** [comment]
Since we use C names for character predicates, maybe we should keep `isupper` and use `toupper` and `tolower`.

File: HISTORY.md [hunk 1]
```diff
@@ -516,7 +516,7 @@ Deprecated or removed
     [#18558], [#19711], [#19712], [#19791], [#19802], [#19931], [#20543], [#20228]).
 
   *  All methods of character predicates (`isalnum`, `isalpha`, `iscntrl`, `isdigit`,
-     `isnumber`, `isgraph`, `islower`, `isprint`, `ispunct`, `isspace`, `isupper`,
+     `isnumber`, `isgraph`, `islowercase`, `isprint`, `ispunct`, `isspace`, `isuppercase`,
```
Comment (JeffBezanson) [inline HISTORY.md]: The history file should not be changed.

**Comment by nalimilan** [comment]
> Since we use C names for character predicates, maybe we should keep isupper and use toupper and tolower.

That's a possibility and it would give shorter identifiers. However the "to" prefix doesn't convey very useful information, and AFAIK we don't use it anywhere else.

**Comment by nalimilan** [comment]
Another advantage of using "case" everywhere is that it's consistent with the `titlecase` function, which can hardly be renamed to `title`.

**Comment by StefanKarpinski** [comment]
We're increasingly deviating from the C names for these and the C names are pretty bad to begin with. I think we should just pick a good set of names for these and be guided by other high-level languages like Python and Ruby (and Swift, etc.).

**Comment by JeffBezanson** [comment]
Ok, I'm fine with the names in this PR. Just throwing out some options.

**Review by JeffBezanson — APPROVED** [review]

**Pull request merged**